content/en/blog/2024/otel-collector-container-log-parser/index.md

```diff
@@ -0,0 +1,176 @@
+[Kubernetes is the leading platform for Collector deployment (80.6%)](/blog/2024/otel-collector-survey/#deployment-scale-and-environment).
+Based on these two facts, we can realize the importance of seamless log
+collection on Kubernetes environments.
+
+Currently, the
+[filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.100.0/receiver/filelogreceiver/README.md)
+is capable of parsing container logs from Kubernetes Pods, but it requires
+[extensive configuration](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/aaa70bde1bf8bf15fc411282468ac6d2d07f772d/charts/opentelemetry-collector/templates/_config.tpl#L206-L282)
+to properly parse logs according to various container runtime formats. This
+configuration complexity can be mitigated by using the corresponding
+[helm chart preset](https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-collector#configuration-for-kubernetes-container-logs).
+However, despite having the preset, it can still be challenging for users to
+maintain and troubleshoot such advanced configurations.
+
+The community has raised the issue of
+[improving the Kubernetes Logs Collection Experience](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25251)
+in the past. One step towards achieving this would be to provide a simplified
+and robust option for parsing container logs without the need for manual
+specification or maintenance of the implementation details. With the proposal
+and implementation of the new
+[container parser](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31959),
+all these implementation details are encapsulated and handled within the
+parser's implementation. Adding to this the ability to cover the implementation
+with unit tests and various fail-over logic indicates a significant improvement
+in container log parsing.
+
+## How container logs look like
+
+First of all we need to quickly recall the different container log formats that
+we can meet out there:
+
+### Docker container logs
+
+`{"log":"INFO: This is a docker log line","stream":"stdout","time":"2024-03-30T08:31:20.545192187Z"}`
+
+### cri-o logs
+
+`2024-04-13T07:59:37.505201169-05:00 stdout F This is a cri-o log line!`
+
+### Containerd logs
+
+`2024-04-22T10:27:25.813799277Z stdout F This is an awesome containerd log line!`
+
+We can notice that cri-o and containerd log formats are quite similar (both
+follow the CRI logging format) but with a small difference in the timestamp
+format.
+
+Consequently, in order to properly handle these 3 different formats we need 3
+different routes of stanza operators as we can see in the
+[container parser operator issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31959).
+
+In addition, the CRI format can provide partial logs which we would like to
+combine them into one at first place:
+
+```console
```

Comment:
~~~suggestion
```console
~~~

@chalin what's the right annotation here, log? text?

Reply:
Right, not `console`. I'd suggest leaving it blank, or use `no-highlight` or `text`.